### ISSUE TITLE
VimwikiGenerateTags links don't handle file hierarchy correctly

### ISSUE THREAD

**Issue by jean** [vimwiki/vimwiki#794]
Create a tag (e.g. `:SomeTag:`), rebuild tags, and in `diary/2020-01-02.wiki` issue the command `:VimwikiGenerateTags`. 

- This results in links like `[[diary/2020-01-02#SomeTag]]`. When that link is followed, vim tries to create `diary/diary/2020-01-02.wiki`. **This is wrong**: 
  - _either_ links should be generated based on the main wiki directory (like `[[/diary/2020-01-02#SomeTag]]`), 
  - _or_ (better) they should be generated relative to the file in which the `:VimwikiGenerateTags` command is issued (`[[2020-01-02#SomeTag]]`). 

This is a follow-on from https://github.com/vimwiki/vimwiki/issues/190.

----

Syntax: default.

## From `.vimrc`:

```vim
let wiki = {}
let wiki.name = 'wiki'
let wiki.path = '~/wiki/'
let wiki.nested_syntaxes = nested_syntaxes
let wiki.auto_tags = 1
```

## The output of `:VimwikiShowVersion`.

Branch: dev
Revision: 5cdeb9a
Date: 2020-01-01 10:58:56 -0700

**Comment by ranebrown** [comment]
Might be able to use `vimwiki#path#relpath(dir, file)` to fix this.

**Comment by tinmarino** [comment]
Fixed #1162
